src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/OptionsSettings.cs

```diff
@@ -8,27 +8,21 @@
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS
 {
-    /// <summary>
-    ///     Provides an implementation of <see cref="IOptionsSettings"/> that delegates onto 
-    /// </summary>
     [Export(typeof(IOptionsSettings))]
     internal class OptionsSettings : IOptionsSettings
     {
         private readonly SVsServiceProvider _serviceProvider;
-        private readonly IProjectThreadingService _threadingService;
-       
+        
         [ImportingConstructor]
-        public OptionsSettings(SVsServiceProvider serviceProvider, IProjectThreadingService threadingService)
+        public OptionsSettings(SVsServiceProvider serviceProvider)
         {
             Requires.NotNull(serviceProvider, nameof(serviceProvider));
-            Requires.NotNull(threadingService, nameof(threadingService));
             _serviceProvider = serviceProvider;
-            _threadingService = threadingService;
         }
 
-        public async Task<T> GetPropertiesValueAsync<T>(string category, string page, string property, T defaultValue)
+        public T GetPropertiesValue<T>(string category, string page, string property, T defaultValue)
         {
-            await _threadingService.SwitchToUIThread();
+            //TODO : VerifyOnUIThread();
```

Comment:
I was looking for a base class that already provided this feature - but given the base class will just import IProjectThreadingService and call this method it's not going to add any value.